### PR TITLE
Full end to end verification of contract fulfillment

### PR DESCRIPTION
**Pull request by j16r**
Main purpose of this PR is to verify that fulfillment is performed in the RunLog test, which involves:

  1. Oracle contract gives permission to chainlink node to fulfill (Oracle.setFulfillmentPermission)
  2. RunLog contract has a fullfillment callback that emits a message
  3. The job now creates a transaction with EthTx
  3. The test asserts that the message is emitted

Refactorings:

  1. ethereum_test now will skip starting services if they are already running
  2. " " " if chainlink is already installed
  3. Assert on differences in counts where possible, so long running chainlink/geth will still pass, i.e. you don't have to wipe anything to get successful tests
  4. chainlink is started in a temporary directory so it won't pollute your local dev environment
  5. Separate env vars for ETH WS vs HTTP

### CODE REVIEW

File: integration/contracts/RunLog.sol [hunk 1]
```diff
@@ -17,7 +18,7 @@ contract RunLog is Chainlinked {
 
   function fulfill(bytes32 _externalId, bytes32 _data)
     public
-    checkChainlinkFulfillment(_externalId)
-  {
+    checkChainlinkFulfillment(_externalId) {
+    emit FulfilledEvent(_data);
```
Comment (se3000) [inline integration/contracts/RunLog.sol]: 👍

File: cmd/local_client_test.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ func TestClient_RunNodeShowsEnv(t *testing.T) {
 	assert.Contains(t, logs, "JSON_CONSOLE: false")
 	assert.Contains(t, logs, "ROOT: /tmp/chainlink_test/")
 	assert.Contains(t, logs, "CHAINLINK_PORT: 6688\\n")
-	assert.Contains(t, logs, "ETH_URL: ws://")
+	assert.Contains(t, logs, "ETH_WS_URL: ws://")
```
Comment (se3000) [inline cmd/local_client_test.go]: This is a breaking change. Can we stick with ETH_URL, for now and maybe move it in a subsequent PR?

File: integration/send_runlog_transaction [hunk 3]
```diff
@@ -34,7 +34,8 @@ const main = async () => {
     'tasks': [
       // 10 seconds to ensure the time has not elapsed by the time the run is triggered
       { 'type': 'Sleep', 'params': { 'until': futureOffset(10) } },
-      { 'type': 'HttpPost', 'params': { 'url': 'http://localhost:6690' } }
+      { 'type': 'HttpPost', 'params': { 'url': 'http://localhost:6690' } },
+      { 'type': 'EthTx', 'params': { 'functionSelector': 'fulfillData(uint256,bytes32)' } }
```
Comment (se3000) [inline integration/send_runlog_transaction]: 💯 This is huge.

File: cmd/local_client_test.go (hunk 2 above)
Comment (j16r) [reply cmd/local_client_test.go]: Okay, I can revert this and use an override in the places where it's used inconsistently.

**Pull request merged**